Model/Indexer/Provider/ChangeLogProvider.php

```diff
@@ -34,27 +34,38 @@
  *
  */
 
-namespace Nosto\Tagging\Util;
+namespace Nosto\Tagging\Model\Indexer\Provider;
 
-use Magento\Framework\App\ObjectManager;
 use Magento\Framework\App\ResourceConnection;
 
-class IndexerChangeLog
+class ChangeLogProvider
```

Comment:
I would call this `ChangeLogService` or `IndexerStatusService`. We can later on add the cleaning of changelogs here as well.